src/core/thread/mle.cpp

```diff
@@ -2308,8 +2308,9 @@ otError Mle::SendAnnounce(uint8_t aChannel, bool aOrphanAnnounce, const Ip6::Add
     ChannelTlv         channel;
     PanIdTlv           panid;
     ActiveTimestampTlv activeTimestamp;
-    Message *          message;
+    Message *          message = NULL;
 
+    VerifyOrExit(netif.GetMac().GetSupportedChannelMask().ContainsChannel(aChannel), error = OT_ERROR_INVALID_ARGS);
```

Comment:
Would it be possible to merge this PR as is since that allows to pass all certification tests immediately, and implement your suggestion later in another?

Reply:
yes, I am fine to go ahead with this PR and update in separate one.

Reply:
@librasungirl,  we already use the channel mask from Active Dataset to determine channels for sending MLE Announcements (e.g., during attach attempts).  

Please see the code here: [mle#1688](https://github.com/openthread/openthread/blob/master/src/core/thread/mle.cpp#L1685) and [mle#2356](https://github.com/openthread/openthread/blob/master/src/core/thread/mle.cpp#L2356).

The change in this commit adds an additional check to ensure the channel is also in "supported channel mask" from MAC layer.

Reply:
@abtink , yes, you are right, I missed that part. Thanks.